crates/providers/src/new.rs

```diff
@@ -0,0 +1,362 @@
+        type Signature = ();
+
+        fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut) {
+            todo!()
+        }
+
+        fn payload_len_for_signature(&self) -> usize {
+            todo!()
+        }
+
+        fn into_signed(
+            self,
+            signature: alloy_primitives::Signature,
+        ) -> alloy_network::Signed<Self, Self::Signature>
+        where
+            Self: Sized,
+        {
+            todo!()
+        }
+
+        fn encode_signed(
+            &self,
+            signature: &alloy_primitives::Signature,
+            out: &mut dyn alloy_primitives::bytes::BufMut,
+        ) {
+            todo!()
+        }
+
+        fn decode_signed(buf: &mut &[u8]) -> alloy_rlp::Result<alloy_network::Signed<Self>>
+        where
+            Self: Sized,
+        {
+            todo!()
+        }
+
+        fn input(&self) -> &[u8] {
+            todo!()
+        }
+
+        fn input_mut(&mut self) -> &mut alloy_primitives::Bytes {
+            todo!()
+        }
+
+        fn set_input(&mut self, data: alloy_primitives::Bytes) {
+            todo!()
+        }
+
+        fn to(&self) -> alloy_network::TxKind {
+            todo!()
+        }
+
+        fn set_to(&mut self, to: alloy_network::TxKind) {
+            todo!()
+        }
+
+        fn value(&self) -> U256 {
+            todo!()
+        }
+
+        fn set_value(&mut self, value: U256) {
+            todo!()
+        }
+
+        fn chain_id(&self) -> Option<alloy_primitives::ChainId> {
+            todo!()
+        }
+
+        fn set_chain_id(&mut self, chain_id: alloy_primitives::ChainId) {
+            todo!()
+        }
+
+        fn nonce(&self) -> u64 {
+            todo!()
+        }
+
+        fn set_nonce(&mut self, nonce: u64) {
+            todo!()
+        }
+
+        fn gas_limit(&self) -> u64 {
+            todo!()
+        }
+
+        fn set_gas_limit(&mut self, limit: u64) {
+            todo!()
+        }
+
+        fn gas_price(&self) -> Option<U256> {
+            todo!()
+        }
+
+        fn set_gas_price(&mut self, price: U256) {
+            todo!()
+        }
+    }
+
+    struct TmpNetwork;
+    impl Network for TmpNetwork {
+        type TxEnvelope = alloy_consensus::TxEnvelope;
+        type ReceiptEnvelope = alloy_consensus::ReceiptEnvelope;
+        type Header = ();
+        type TransactionRequest = TxLegacy;
+        type TransactionResponse = ();
+        type ReceiptResponse = ();
+        type HeaderResponse = ();
+    }
+
+    fn init_tracing() {
+        let _ = tracing_subscriber::fmt::try_init();
+    }
+
+    #[tokio::test]
+    async fn test_send_tx() {
+        init_tracing();
+
+        let anvil = alloy_node_bindings::Anvil::new().block_time(1u64).spawn();
+        let url = anvil.endpoint().parse().unwrap();
+        let http = Http::<Client>::new(url);
+        let provider = RootProvider::<TmpNetwork, _>::new(RpcClient::new(http, true));
```

Comment:
we should sugar this to `Provider::new(url)` and have it work for all HTTP/WS/IPC etc.